pkg/performanceprofile/controller/performanceprofile/components/tuned/tuned.go

```diff
@@ -40,8 +41,17 @@ const (
 	templatePerformanceProfileName          = "PerformanceProfileName"
 )
 
-func new(name string, profiles []tunedv1.TunedProfile, recommends []tunedv1.TunedRecommend) *tunedv1.Tuned {
-	return &tunedv1.Tuned{
+func OptionsFromPerformanceProfile(pp *performancev2.PerformanceProfile) components.TunedOptions {
```

Comment:
This is more a question than a comment, why it's better to return a struct than a pointer here?

Reply:
In this case is debatable indeed. I tend to use return by value (meaning return structs, not pointer to) by default because I like to minimize the mutability side.
Using values make it obvious we work on copies, there's (the promise of: members can be pointers!) no mutability, which simplifies the reasoning.
For small structs the performance loss is negligible, and in general optimization should be data-driven, the code should default to readability.

But again: in this case is mostly habit, returning a pointer wouldn't have raised eyebrows.

Reply:
Ack. Thank you for the explanation